File: mycollab-web/src/main/scala/com/mycollab/module/project/event/BugEvent.scala
package com.mycollab.module.project.event

import com.mycollab.events.ApplicationEvent
import com.mycollab.module.tracker.domain.criteria.BugSearchCriteria

/**
  * @author MyCollab Ltd.
  * @since 5.0.3
  */
object BugEvent {
  
  class GotoAdd(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}
  
  class GotoEdit(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}
  
  class GotoRead(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}
  
  class SearchRequest(source: AnyRef, data: BugSearchCriteria) extends ApplicationEvent(source, data) {}
  
  class BugChanged(source: AnyRef, data: Integer) extends ApplicationEvent(source, data) {}
  
}

File: mycollab-web/src/main/java/com/mycollab/module/project/event/BugEvent.kt
package com.mycollab.module.project.event

import com.mycollab.vaadin.event.ApplicationEvent
import com.mycollab.module.tracker.domain.criteria.BugSearchCriteria

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
object BugEvent {
    class GotoAdd(source: Any, val data: Any?) : ApplicationEvent(source)

    class GotoEdit(source: Any, val data: Any?) : ApplicationEvent(source)

    class GotoRead(source: Any, val data: Any?) : ApplicationEvent(source)

    class SearchRequest(source: Any, data: BugSearchCriteria) : ApplicationEvent(source)

    class BugChanged(source: Any, data: Int) : ApplicationEvent(source)
}